src/steps/fetch-404.js

```diff
@@ -39,18 +39,22 @@ export default async function fetch404(state, req, res) {
   }
 
   // set 404 keys in any case
-  const pathKey = await getPathKey(state);
   // provide either (prefixed) preview or (unprefixed) live content keys
   const contentKeyPrefix = partition === 'preview' ? 'p_' : '';
-  const keys = [
-    `${contentKeyPrefix}${pathKey}`,
-    `${contentKeyPrefix}${contentBusId}`,
-    `${ref}--${repo}--${owner}_404`,
-    `${ref}--${repo}--${owner}_code`,
-  ];
+  const keys = [];
+  if (state.content.sourceBus === 'code') {
+    keys.push(await computeCodePathKey(state));
+  } else {
+    const contentPathKey = await computeContentPathKey(state);
+    // provide either (prefixed) preview or (unprefixed) live content keys
+    keys.push(`${contentKeyPrefix}${contentPathKey}`);
+    keys.push(`${contentKeyPrefix}${contentBusId}`);
+  }
+  keys.push(`${ref}--${repo}--${owner}_404`);
+  keys.push(`${ref}--${repo}--${owner}_code`);
```

Comment:
I don't think this is correct. we always need both keys, as the missing resource can be added as code or as content.

Reply:
ok, but that was never the case. only the content key was added.